src/mymemory.c

```diff
@@ -227,5 +227,80 @@ void *sw_memccpy_custom(
     return 0;
 }
 
+/**
+@brief Wrapper function to `sw_memccpy_custom()` which copies data of a
+    but this function also removes the repetitve action of decreasing
+    the available size left in the allocated string location
+
+@note This function uses the compiler macro '__restrict' instead of simply
+'restrict' due to C++ standards not supporting it, so '__restrict' is
+compatible in Clang and GCC
+
+@param[in,out] charPtr Pointer holding the location of the writing
+start location of the string
+@param[in] str Array of characters to copy/concatenate into `charPtr`
+@param[in] c Target character which, upon finding, is one of the stopping
+condiditions
+@param[in,out] n The number of bytes to copy from src to dest, and is the
+second stopping condition
+
+@return A flag specifying if the buffer we are copying/concatenating into
+is full
+ */
+Bool sw_memccpy_inc(

```

Comment:
sw_memccpy_inc(): one more question: why is `*endPtr = '\0';` separated out in sw_memccpy_report()? Why not make it part of sw_memccpy_inc() so that all string/buffer manipulation is together? The code already checks for failure. For instance,
```
Bool sw_memccpy_inc(..., char *endPtr, ...) {
...
if (isnull(resPtr)) {
    if (!isnull(endPtr)) *endPtr = '\0';
} else {
    ...
}
...
}
```